runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecord.java

```diff
@@ -193,8 +212,33 @@ void serialize(ByteBuf b) {
                     b.writeInt(length);
                     b.writerIndex(lengthIndex + length + 4);
                 });
-        // including the size of serializedSize itself.
-        serializedSize = b.writerIndex() - startPos + 4;
+        // Calculate the serialized size for garbage marking purpose.
+        serializedSize = b.writerIndex() - startPos;
+        b.writerIndex(sizeIndex);
         b.writeInt(serializedSize);
+        b.writerIndex(startPos + serializedSize);
+    }
+
+    /**
+     * Get the bytes of one {@link SMRRecord} from a larger byte buffer
+     * and transfer it to a new byte array.
+     *
+     * @param b byte buffer to transfer from
+     * @return a new byte array containing the transferred bytes
+     */
+    static byte[] transfer(ByteBuf b) {
```

Comment:
This name is confusing. Probably toByte is a better name.

Reply:
Changed name to slice, as it's not only converting to bytes, it slices from a larger buffer.